src/libraries/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs

```diff
@@ -36,22 +34,15 @@ public SendPacketsAsync(ITestOutputHelper output)
                 buffer[i] = (byte)(i % 255);
             }
 
-            try
-            {
-                _log.WriteLine("Creating file {0} with size: {1}", TestFileName, s_testFileSize);
-                using (FileStream fs = new FileStream(TestFileName, FileMode.CreateNew))
-                {
-                    fs.Write(buffer, 0, buffer.Length);
-                }
-            }
-            catch (IOException)
-            {
-                // Test payload file already exists.
-                _log.WriteLine("Payload file exists: {0}", TestFileName);
-            }
+            _tempFile = TempFile.Create(buffer);
```

Comment:
One good aspect of the current approach is that Helix _can_ cleanup on test or other failures. Since the run directory is unique this is very easy. I don't know if Helix actually does: cc @MattGal 
Using `TempFile.Create` has potential of filling up temp space. 
File name starting with '\' seems weird but I don't know if that is relevant.

Reply:
Unless there is a hang or a crash, `TempFile.Dispose()` should delete the file after every test run:
https://github.com/dotnet/runtime/blob/68d0d028636d7ccf791c61e732dd955ff28b2f1c/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendPacketsAsync.cs#L41-L44

If we want to be more robust, it should be a feature request towards Helix, so they cleanup `Path.GetTempPath()` after the test process finished. [Plenty of BCL tests](https://github.com/dotnet/runtime/search?q=TempFile) use the `TempFile` util, I see no point keeping a custom approach for `SendPacketsAsync`.